components/director/internal/domain/formation/service.go

```diff
@@ -47,9 +59,18 @@ type uidService interface {
 
 //go:generate mockery --exported --name=automaticFormationAssignmentService --output=automock --outpkg=automock --case=underscore --disable-version-string
 type automaticFormationAssignmentService interface {
-	Create(ctx context.Context, in model.AutomaticScenarioAssignment) (model.AutomaticScenarioAssignment, error)
 	GetForScenarioName(ctx context.Context, scenarioName string) (model.AutomaticScenarioAssignment, error)
-	Delete(ctx context.Context, in model.AutomaticScenarioAssignment) error
+}
+
+//go:generate mockery --exported --name=automaticFormationAssignmentRepository --output=automock --outpkg=automock --case=underscore --disable-version-string
+type automaticFormationAssignmentRepository interface {
+	Create(ctx context.Context, model model.AutomaticScenarioAssignment) error
+	ListForTargetTenant(ctx context.Context, tenantID string, targetTenantID string) ([]*model.AutomaticScenarioAssignment, error)
+	GetForScenarioName(ctx context.Context, tenantID, scenarioName string) (model.AutomaticScenarioAssignment, error)
+	ListAll(ctx context.Context, tenantID string) ([]*model.AutomaticScenarioAssignment, error)
+	List(ctx context.Context, tenant string, pageSize int, cursor string) (*model.AutomaticScenarioAssignmentPage, error)
```

Comment:
Same thing as the previous comment, while I was moving the implementation I seem to have used the full interface that was used in the previous ASA repo interface, out of these in the formations service the only used ones are `Create`, `DeleteForTargetTenant` and `DeleteForScenarioName`, because the corresponding service layer methods were moved here. I will remove unused interface methods.